base/show.jl

```diff
@@ -2718,7 +2718,7 @@ function type_depth_limit(str::String, n::Int; maxdepth = nothing)
         end
         prev = di
     end
-    return String(take!(output))
+    return unsafe_takestring!(output)
```

Comment:
Isn't it clearly safe because the `output` object does not escape?

Reply:
That isn't clear to me. Are `write(output, c)` and `write(output, c.del)` guaranteed not to escape `output`? Testing that seems out of scope for this PR.

Reply:
Yes, because Base owns the methods `write(::IOBuffer, ::Char)` and `write(::IOBuffer, ::SubString{String})`.

(The same requirement is made in many other uses of `unsafe_takestring!` and `unsafe_take!` in Base … they are used whenever a function writes to an `IOBuffer` that is allocated internally to the function.  It seems strange to quibble about them here.)

Reply:
(There is a legitimate worry about functions like `sprint` that were already using `_unsafe_take!`, but were passing it to user-defined methods that might conceivably save the buffer.)

Reply:
Okay, that makes sense.